coverage/artifacts/artifacts.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package artifacts is responsible for generating and structuring artifacts
+// directory
+package artifacts
+
+import (
+	"path"
+)
+
+const (
+	//CovProfileCompletionMarker is the file name of the completion marker for coverage profiling
+	CovProfileCompletionMarker = "profile-completed"
+	junitXmlForTestgrid        = "junit_bazel.xml"
+	//LineCovFileName is the file name for line coverage html
+	LineCovFileName = "line-cov.html"
+)
+
+//Artifacts stores information about artifacts directory and files in it
+type Artifacts struct {
+	directory      string
+	profileName    string
+	keyProfileName string
+	covStdoutName  string
+}
+
+//New Artifact object
+func New(directory, profileName, keyProfileName, covStdoutName string) *Artifacts {
```

Comment:
Actually some of the fields are not supposed to be read by external packages either. External package should only use the exported methods, which are usually not as simple as a getter of the fields

Reply:
for these fields all of the setters are trivial, the combined getters that join paths are orthogonal. it's unidiomatic to have trivial getters and setters to say the least. I don't see any invariants being enforced here, and in go that's normally done:

- at New() time
- alternatively in some delayed validation call. possibly before usage.

getters and setters are normally only used for non-trivial derivation of fields.

Reply:
Do you mean to give the final value for all the fields at the time of New and make all fields public? Don't we want to prevent the value from being modified later?

Reply:
If we have a setter then we're not preventing that are we? ...
the other getters are not for values, they do some computation so of course those stay getters.

Reply:
There is no setter for any fields other than directory, because I don't want to let those values be modifiable. I don't think I fully get what your suggestion is

Reply:
@BenTheElder